src/scratchtocatrobat/converter/mediaconverter.py

```diff
@@ -97,76 +96,94 @@ def convert(self, progress_bar = None):
+        self.rename_media_files_and_copy()
 
-                if is_unconverted:
-                    unconverted_media_resources.append(resource_info)
-                elif progress_bar != None and sound_src_path not in converted_media_resources_paths:
-                    # update progress bar for all those media files that don't have to be converted
-                    progress_bar.update(ProgressType.CONVERT_MEDIA_FILE)
-                    converted_media_resources_paths.add(sound_src_path)
+        # delete converted png files -> only temporary saved
+        for media_file_to_be_removed in converted_media_files_to_be_removed:
+            os.remove(media_file_to_be_removed)
 
+    def setup_costume_info(self, defined_scratch_object, all_used_resources, unconverted_media_resources,
+                           converted_media_resources_paths, progress_bar):
+        threads = []
+        for costume_info in defined_scratch_object.get_costumes():
+            costume_dict = self.get_info(costume_info[JsonKeys.COSTUME_MD5], True)
+
+            assert os.path.exists(costume_dict["costume_src_path"]), "Not existing: {}".format(costume_dict["costume_src_path"])
+            assert costume_dict["file_ext"] in {".png", ".svg", ".jpg", ".gif"}, \
+                "Unsupported image file extension: %s" % costume_dict["costume_src_path"]
+            ispng = costume_dict["file_ext"] == ".png"
+            is_unconverted = costume_dict["file_ext"] == ".svg"
+            self.setup_resource_info_dict(costume_dict["costume_file_name"], costume_dict["costume_src_path"], is_unconverted, costume_info,
+                                          all_used_resources, unconverted_media_resources, converted_media_resources_paths,
+                                          progress_bar, threads, ispng, True)
+        for thread in threads: thread.start()
+        for thread in threads: thread.join()
+
+    def setup_sound_info(self, defined_scratch_object, all_used_resources, unconverted_media_resources, converted_media_resources_paths, progress_bar):
+        for sound_info in defined_scratch_object.get_sounds():
+            sound_dict = self.get_info(sound_info[JsonKeys.SOUND_MD5])
+            assert os.path.exists(sound_dict["sound_src_path"]), "Not existing: {}".format(sound_dict["sound_src_path"])
+            assert sound_dict["file_ext"] in {".wav", ".mp3"}, "Unsupported sound file extension: %s" % sound_dict["sound_src_path"]
+            is_unconverted = sound_dict["file_ext"] == ".wav" and not wavconverter.is_android_compatible_wav(sound_dict["sound_src_path"])
+
+            self.setup_resource_info_dict(sound_dict["sound_file_name"], sound_dict["sound_src_path"], is_unconverted, sound_info,
+                                          all_used_resources, unconverted_media_resources, converted_media_resources_paths,
+                                          progress_bar, [])
+
+    def get_info(self, file_name, is_costume=False):
```

Comment:
This function basically contains the same code twice, the only difference being the naming of dictionary keys. Alternatively, more neutral names could be used to describe both, costume and sound files, and if necessary a 'type' field could be used to remember the type to the file.